Makes models

Tungtt2020 opened the pull request.

aleboj commented on openapi.yaml:
```diff
@@ -9,36 +9,63 @@ info:
     get:
       operationId: GetMakes
-      description: Returns a list of makes sorted alphabetically.
+      description: Returns a sorted list of makes for a vehicle category
```
missing `.` at the end of the sentence

aleboj commented on openapi.yaml:
```diff
@@ -9,36 +9,63 @@ info:
+    get:
+      tags:
+        - Model
+      description: Returns a sorted list of models by makeKey and vehicleCategory
```
missing `.` at the end of the sentence

aleboj commented on openapi.yaml:
```diff
@@ -9,36 +9,63 @@ info:
+        - $ref: "#/components/parameters/sizeQueryParam"
+      responses:
+        "200":
+          description: Models for a makeKey and vehicle category returned
```
missing `.` at the end of the sentence

aleboj commented on openapi.yaml:
```diff
@@ -61,24 +88,113 @@ components:
+          type: string
+          example: "a4-allroad"
+        name:
+          description: Display name of the model
```
missing `.` at the end of the sentence

aleboj commented on openapi.yaml:
```diff
@@ -61,24 +88,113 @@ components:
+        empty:
+          type: boolean
+          example: false
+        numberOfElements:
+          description: Actual number of elements in a page
+          type: integer
+          format: int32
+          example: 18
+        size:
+          description: Size of a page
+          type: integer
+          format: int32
+          example: 20
+        totalElements:
+          description: Total number of elements
+          type: integer
+          format: int64
+          example: 138
+        first:
+          description: Is first page
+          type: boolean
+          example: false
+        last:
+          description: Is last page
+          type: boolean
+          example: true
+        number:
+          description: Page number
+          type: integer
+          format: int32
+          example: 6
+        totalPages:
+          description: Total number of pages
+          type: integer
+          format: int32
+          example: 7
```
missing `.` at the end of the sentence

aleboj commented on openapi.yaml:
```diff
@@ -61,24 +88,113 @@ components:
         minimum: 1
         maximum: 2000
-        default: 25
+        default: 20
```
isn't it 25?

Tungtt2020 replied:
no, it is 20 by default

milanovici reviewed (approved):
👍

lukicsasa reviewed (approved):
Looks good to me

Tungtt2020 commented:
i also aligned nested schema name to swagger validator, removed suffix dto, adjusted default page size example, extracted enum types to reusable objects.
checked with swagger UI and got valid badge.
![image](https://github.com/smg-automotive/autoscout24-api-docs/assets/60682061/5b598e4b-5ea4-40d0-9146-99290d8192ae)
![image](https://github.com/smg-automotive/autoscout24-api-docs/assets/60682061/1968faca-5705-487a-a396-d96e719c16bc)
